src/coreComponents/constitutive/solid/DuvautLionsSolid.hpp

```diff
@@ -0,0 +1,191 @@
+ * Copyright (c) 2019-     GEOSX Contributors
+ * All rights reserved
+ *
+ * See top level LICENSE, COPYRIGHT, CONTRIBUTORS, NOTICE, and ACKNOWLEDGEMENTS files for details.
+ * ------------------------------------------------------------------------------------------------------------
+ */
+
+
+/**
+ * @file DuvautLionsSolid.hpp
+ * @brief This class implements Duvaut-Lions viscoplasticity model
+ *
+ */
+
+#ifndef GEOSX_CONSTITUTIVE_SOLID_DUVAUTLIONSSOLID_HPP_
+#define GEOSX_CONSTITUTIVE_SOLID_DUVAUTLIONSSOLID_HPP_
+
+#include "constitutive/solid/SolidBase.hpp"
+#include "ElasticIsotropic.hpp"
+#include "DruckerPrager.hpp"
+#include "DruckerPragerExtended.hpp"
+#include "SolidModelDiscretizationOpsFullyAnisotroipic.hpp"
+#include "SolidModelDiscretizationOpsIsotropic.hpp"
+#include "LvArray/src/tensorOps.hpp"
+
+namespace geosx
+{
+namespace constitutive
+{
+
+// DAMAGE MODEL UPDATES
+//
+// NOTE: This model uses the m_newStress array to represent the stress in a
+//       "non-viscous" elasto-plastic solid.  We then scale the results
+//       by the damage factor whenever the true stress is requested through an update
+//       function.  The developer should be very cautious if accessing the stress
+//       directly through an arrayView, as it does not represent the true stress.
+//
+
+template< typename UPDATE_BASE >
+class DuvautLionsSolidUpdates : public UPDATE_BASE
+{
+public:
+  template< typename ... PARAMS >
+  DuvautLionsSolidUpdates( real64 const & relaxationTime,
+                           PARAMS && ... baseParams ):
+    UPDATE_BASE( std::forward< PARAMS >( baseParams )... ),
+    m_relaxationTime( relaxationTime )
+  {}
+
+  // using DiscretizationOps = typename UPDATE_BASE::DiscretizationOps;
+  using DiscretizationOps = typename UPDATE_BASE::DiscretizationOps; // TODO: typo in anistropic (fix in DiscOps PR)
+
+  using UPDATE_BASE::smallStrainUpdate;
+  //using UPDATE_BASE::smallStrainUpdate_ElasticOnly;
+  using UPDATE_BASE::saveConvergedState;
+
+  using UPDATE_BASE::viscousStateUpdate;
+
+  using UPDATE_BASE::m_disableInelasticity;
+
+  GEOSX_HOST_DEVICE
+  virtual void smallStrainUpdate( localIndex const k,
+                                  localIndex const q,
+                                  real64 const & timeIncrement,
+                                  real64 const ( &strainIncrement )[6],
+                                  real64 ( & stress )[6],
+                                  real64 ( & stiffness )[6][6] ) const override
+  {
+    real64 trialStress[6];   // Trial stress (elastic predictor)
+    real64 elasticStiffness[6][6];  //Elastic stiffness
+    real64 timeRatio = 1.0 / (1.0 + timeIncrement / m_relaxationTime);
+
+    for( localIndex i=0; i<6; ++i )
+    {
+      trialStress[i] = stress[i];
+    }
+
+    if( m_disableInelasticity )
+    {
+      return;
+    }
+
+    //Get trial stress and elastic stiffness by disabling inelasticity
+    // m_solidUpdate.m_disableInelasticity = true;
```

Comment:
Do you still need to disable inelasticity here? Or `smallStrainUpdate_ElasticOnly` will do it automatically? If it will, you can remove the commented code here and below.

Reply:
Done